### PR TITLE
BUILD: Various macOS build fixes for Tiger/Leopard

### PR DESCRIPTION
**Pull request by dwatteau**
ScummVM 2.2.0 was the last official release for Mac PowerPC; it still builds if you set up a C++11 toolchain for it, but you need dynamic plugins now (see PR #3877) because a single binary with all engines is too big for the PowerPC Mach-O linker.

You also need this small set of fixes:

* don't use the `-Os -mlongcall` workaround on Mac PPC anymore: the engines are too big for a single Mach-O PowerPC binary, now (ELF has better workarounds for this). I tried some other linker or compiler workarounds, but it wouldn't be enough, or it would cause internal linker errors. So I suggest relying on dynamic plugins for a full build in this particular setup (see PR #3877 for this).
    * Going back to default GCC flags could also mean fewer GCC bugs on this old platform with its old toolchain.
* don't run `codesign` when targeting Leopard or Tiger: Tiger didn't have it at all, and Leopard didn't have the `--deep` flag. The OS didn't really care for that yet, anyway.
* work around limited `sed -i''` before macOS 10.9: it's just impossible to have a `sed -i''` syntax that will be accepted by older BSD sed, newer BSD sed, and GNU sed ([source](https://stackoverflow.com/questions/4247068/#4247319)). So here, always use a suffix, and delete the backup files afterward (otherwise, the bundle will embed some strange `-e` files).
    * (There are other calls to `sed -i''` but the Tiger/Leopard build doesn't hit them.)
* reject GNU Make 3.80 and older: that's the default version on macOS Tiger, but it's very old and it has some bugs and parsing issues, so it shouldn't be used.
    * It's easy to have at least GNU Make 3.81 everywhere, and we know that 3.81 is OK because that's what Apple still ships on modern macOS (because of GPLv3).
    * Parsing `$(MAKE_VERSION)` is quite inconvenient and not very reliable, so it's usually easier to look for a feature which appeared in the minimum version you require (here, the `.FEATURES` variable). Putting this at the top of the main Makefile will also "help" BSD `make` fail early on (since we require GNU Make but don't use `GNUmakefile`).
* remove old hacks for CoreAudio on macOS 10.2 and 10.3. I'm pretty sure the rest of the macOS code didn't compile on macOS 10.2/10.3 for a long time, and there's no C++11 toolchain before macOS 10.4 anyway.

Tested on macos Tiger, Leopard and Monterey.

### CODE REVIEW

**Comment by raziel-** [comment]
Nice to have another PPC platform targeted, maybe we can help out with BE bugs in the future?

**Comment by dwatteau** [comment]
> Nice to have another PPC platform targeted, maybe we can help out with BE bugs in the future?

Thanks, yeah there's still a retro PPC audience, so if you take all Mac PPC + PS3 + Wii + AmigaOS users you get a higher chance of finding and fixing some bugs here and there. My old PowerBook was helpful in fixing big-endian bugs in Duckman, Full Pipe, Galador, Blazing Dragons… and I think your AmigaOS work helped Grim Fandango work out of the box there :)

**Comment by raziel-** [comment]
Yep :-)

Just you don't go anywhere ;-)

**Comment by bonki** [comment]
I have an old PPC iMac G4 lying around in the attic which I have been meaning to resurrect for ScummVM for years now. Maybe it's time to do some spring cleaning after all :-)

**Comment by dwatteau** [comment]
> I have an old PPC iMac G4 lying around in the attic which I have been meaning to resurrect for ScummVM for years now. Maybe it's time to do some spring cleaning after all :-)

Ha, thanks ;) I've put a 2.6.0git build (with default engines) here, if you're interested:
https://github.com/dwatteau/scummvm/releases

I need to submit something for the toolchain, too…

**Review by criezy — APPROVED** [review]
Overall this looks good.

The only commit I am not sure about is df5aedc9bbac728865a771d725748ec88db64b54 as it potentially touches other platforms. But I assume that if make 3.80 has bugs that make it unusable on MacOS X, that version would be unusable on other platforms as well. And it is indeed a very old version.

This also reminded me of that ticket I created 12 years ago: https://bugs.scummvm.org/ticket/9133

**Comment by criezy** [comment]
I will wait a bit longer before merging in case others want to review the changes.

**Comment by dwatteau** [comment]
> Overall this looks good.
> 
> The only commit I am not sure about is [df5aedc](https://github.com/scummvm/scummvm/commit/df5aedc9bbac728865a771d725748ec88db64b54) as it potentially touches other platforms. But I assume that if make 3.80 has bugs that make it unusable on MacOS X, that version would be unusable on other platforms as well. And it is indeed a very old version.
> 
> This also reminded me of that ticket I created 12 years ago: https://bugs.scummvm.org/ticket/9133

Yeah, basically GNU Make 3.80 is from 2002 while GNU Make 3.81 is from 2006, so requiring that upgrade seems quite reasonable, to me.

One of the problems with GNU Make 3.80 is that 

```sh
else ifeq "$(CUR_BRANCH)" ""
```

will cause a parsing issue, for example. We could change that line, but it has been there for more than 10 years, so it looks like that this version has been untested for a very long time. So I think it's not safe to allow someone build with it and possibly have some strange results. The NEWS file for GNU Make 3.81 mentions various incompatibilities with the previous version, so it really doesn't look safe to me to continue supporting anything older if no port ever test it.

GNU Make 3.81 is well tested "thanks" to base macOS sticking with it, and since it's the last GPLv2 version, other systems are potentially stuck with that one by default too.

File: configure [hunk 1]
```diff
@@ -2652,17 +2652,11 @@ case $_host_os in
...
 	darwin*)
-		# Pass -mlongcall to gcc so that it emits long calls
-		# which will allow for calls larger than 32MB. The linker
-		# will discard the calls if they are not needed, but we
-		# need to ensure the compiler emits them in the first place.
-		# Also the executable has grown to a size where using -Os is necessary to avoid a
-		# 'virtual memory exhausted' error when running the executable.
 		case $_host_cpu in
 		powerpc*)
-			append_var CFLAGS "-mlongcall"
```
Comment (lephilousophe) [inline configure:2663]: Why did you remove the `-mlongcall` option? Is it harmful?
It seems to help on AmigaOS4 where a dynamic modules build is done too.

Comment (dwatteau) [reply configure:2663]: When switching to dynamic engine plugins on Mac PPC, the main (static) `scummvm` binary is now just 15MB (it was 52MB for ScummVM 2.2.0), so the `-mlongcall` flag for "calls larger than 32MB" appears unnecessary again, and I think we have enough room to avoid hitting this limit soon.

Adding it would mean that "all such calls take up one or two extra instructions" and I thought that it's better to keep the compiler/platform defaults whenever possible. Especially as the toolchain is a bit fragile there (the old Mach-O linker in particular is quite buggy/limited; and contrary to GNU binutils `ld`, nobody maintains the PPC Mach-O `ld` anymore; so the more you follow common usage the less you hit corner-case bugs).

As for AmigaOS4, I think it's maintained a bit differently, for example some toolchain flags appear to be used/kept "just in case" there. I think both approaches are fine for less-common platforms, it's just that the Mac PPC toolchain is not ELF and the Mach-O tools there are old, buggy and unmaintained, so I'd rather avoid non-default options if I don't need them and if they'd impact the Mach-O tools.

Plus, having the build fail at the very last linking stage after a 5-hour build because the `.o` files were built with something a bit more obscure that the Mach-O linker has troubles with is not fun ;) (and yeah I've lost some days to that in my previous attempts).

**Comment by raziel-** [comment]
@dwatteau 

>Plus, having the build fail at the very last linking stage after a 5-hour build because the .o files were built with something a bit more obscure that the Mach-O linker has troubles with is not fun ;) (and yeah I've lost some days to that in my previous attempts).

I hear you...
Lost a few days myself hunting for .so load bugs

**Comment by criezy** [comment]
Merged manually to squash bd4bb98 with df5aedc.
